gauge-ts/src/processors/params/PrimitiveParser.ts

```diff
@@ -27,8 +27,12 @@ export class PrimitiveParser implements ParameterParser {
   }
 
   private convertToNumber(value: string): number | undefined {
-    const num = Number.parseFloat(value);
-    return Number.isNaN(num) ? undefined : num;
+    const trimmedValue = value.trim();
+    if (/^-?\d+(\.\d+)?$/.test(trimmedValue)) {
```

Comment:
I thought of using a regex but wanted to avoid it because it seemed overkill, but I'm afraid we don't have a choice.
I came up during my investigations with this regex: `/^[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?$/` to handle extreme case where users would use exponential notations (e.g. `2e-32`). Seems unlikely, but...well better safe than sorry 😄

Reply:
> What if Gauge allowed the "special parameter" notation to specify a plain string like <string:0123456> and then it would bypass this? 

Thanks for the suggestion! The idea of using a special parameter like `<string:0123456>` makes sense. However, it's best to keep Gauge specs as readable and close to markdown as possible. This is to have that clear separation and make the reports readable to people who are not that technical.

Reply:
@anthony-legay pushed a simpler approach refer 910e6144d577

Is there a way you can test this locally?